Automattic/wp-calypso
# Instrument blocking of stats loader

Some Chrome plugins like uBlock have default blocklists that prevent Chrome from loading stats.wp.com/w.js.

In order to estimate the loss of tracking on our funnels, this change logs how many users are affected by loading a /nostats.js file which logs a warning and sends a pixel to Tracks with the event name `calypso_nostats`

In my investigations I estimated that around 12-15% of browsers were blocking stats.wp.com or pixel.wp.com, but this will hopefully give us better data so we can adjust our analysis.

*This is intended as a proof of concept for discussion*

Note that some changes in here are because files violated lint rules, e.g. changes from var to let/const.

To test:

- install uBlock Origin in Chrome
- Load calypso.localhost:3000
- Note that calls to stats.wp.com and pixel.wp.com are blocked
- Observe call to /nostats.js
- Should have a console message indicating stats are blocked
- Server should send a pixel request with `_en: 'calyspo_nostats'`

It would be great if reviewers could test in a variety of browsers with a variety of blockers.
